File: src/main/java/cn/aixcyi/plugin/tinysnake/ui/DunderAllOptimizer.java
package cn.aixcyi.plugin.tinysnake.ui;

import cn.aixcyi.plugin.tinysnake.storage.DunderAllOptimization;
import com.intellij.openapi.ui.DialogWrapper;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.awt.*;

import static cn.aixcyi.plugin.tinysnake.Zoo.message;

/**
 * {@code __all__} 格式化设置面板。
 *
 * @author <a href="https://github.com/aixcyi">砹小翼</a>
 */
public class DunderAllOptimizer extends DialogWrapper {

    public DunderAllOptimization.State state;

    private JPanel       contentPanel;
    private ButtonGroup  groupOrder;
    private JRadioButton radioAlphabetOrder;
    private JRadioButton radioCharOrder;
    private JRadioButton radioAppearanceButton;
    private JCheckBox    checkboxUseSingleQuote;
    private JCheckBox    checkboxEndsWithComma;
    private JCheckBox    checkboxLineByLine;

    public DunderAllOptimizer() {
        super(true);
        setResizable(false);
        setTitle(message("command.OptimizeDunderAll"));
        init();
        load();
    }

    private void load() {
        state = DunderAllOptimization.getInstance().getState();
        switch (state.getMySequenceOrder()) {
            case APPEARANCE -> groupOrder.setSelected(radioAppearanceButton.getModel(), true);
            case ALPHABET -> groupOrder.setSelected(radioAlphabetOrder.getModel(), true);
            case CHARSET -> groupOrder.setSelected(radioCharOrder.getModel(), true);
        }
        checkboxUseSingleQuote.setSelected(state.isUseSingleQuote());
        checkboxEndsWithComma.setSelected(state.isEndsWithComma());
        checkboxLineByLine.setSelected(state.isLineByLine());
    }

    private void save() {
        state.setMySequenceOrder(radioCharOrder.isSelected()
                ? DunderAllOptimization.Order.CHARSET
                : radioAlphabetOrder.isSelected()
                ? DunderAllOptimization.Order.ALPHABET
                : DunderAllOptimization.Order.APPEARANCE
        );
        state.setUseSingleQuote(checkboxUseSingleQuote.isSelected());
        state.setEndsWithComma(checkboxEndsWithComma.isSelected());
        state.setLineByLine(checkboxLineByLine.isSelected());
    }

    @Override
    public boolean showAndGet() {
        if (super.showAndGet()) {
            this.save();
            return true;
        }
        return false;
    }

    @Override
    protected @Nullable JComponent createCenterPanel() {
        JPanel dialogPanel = new JPanel(new BorderLayout());
        dialogPanel.add(contentPanel, BorderLayout.CENTER);
        return dialogPanel;
    }
}

File: src/main/kotlin/cn/aixcyi/plugin/tinysnake/ui/DunderAllOptimizer.kt
package cn.aixcyi.plugin.tinysnake.ui

import cn.aixcyi.plugin.tinysnake.storage.DunderAllOptimization
import cn.aixcyi.plugin.tinysnake.util.IOUtil.message
import cn.aixcyi.plugin.tinysnake.util.mnemonic
import com.intellij.openapi.ui.DialogWrapper
import com.intellij.ui.dsl.builder.bind
import com.intellij.ui.dsl.builder.bindSelected
import com.intellij.ui.dsl.builder.panel

/**
 * `__all__` 格式化设置面板。
 *
 * @author <a href="https://github.com/aixcyi">砹小翼</a>
 */
class DunderAllOptimizer : DialogWrapper(true) {

    val state = DunderAllOptimization.getInstance().state

    init {
        isResizable = false
        title = message("command.OptimizeDunderAll")
        setOKButtonText(message("button.OK.text"))
        setCancelButtonText(message("button.Cancel.text"))
        super.init()
    }

    override fun createCenterPanel() = panel {
        group(message("label.DunderAllOptimizerOrdering.text")) {
            buttonsGroup {
                row {
                    radioButton(
                        message("radio.DunderAllOrderByAppearance.text"), DunderAllOptimization.Order.APPEARANCE
                    ).mnemonic()
                }
                row {
                    radioButton(
                        message("radio.DunderAllOrderByAlphabet.text"), DunderAllOptimization.Order.ALPHABET
                    ).mnemonic()
                }
                row {
                    radioButton(
                        message("radio.DunderAllOrderByCharacter.text"), DunderAllOptimization.Order.CHARSET
                    ).mnemonic()
                }
            }.bind(state::mySequenceOrder)
        }
        group(message("label.DunderAllOptimizerOptions.text")) {
            row {
                checkBox(message("checkbox.DunderAllEndsWithComma.text"))
                    .bindSelected(state::isEndsWithComma)
                    .mnemonic()
            }
            row {
                checkBox(message("checkbox.DunderAllUseSingleQuote.text"))
                    .bindSelected(state::isUseSingleQuote)
                    .mnemonic()
            }
            row {
                checkBox(message("checkbox.DunderAllLineByLine.text"))
                    .bindSelected(state::isLineByLine)
                    .mnemonic()
            }
        }
    }
}